gui/default/syncthing/core/syncthingController.js

```diff
@@ -14,10 +14,63 @@ angular.module('syncthing.core')
 
         function initController() {
             LocaleService.autoConfigLocale();
+            checkLoginAndStart();
+        }
+
+        function checkLoginAndStart() {
+            $http.get(urlbase + '/login').success(function() {
+                $('#logout').html(`
+                                    <span class="fas fa-fw fa-sign-out"></span>&nbsp;
+                                    <span translate>Logout</span>
+                                `)
+                // We are logged in, or don't need to log in
+                loginSuccessfull()
+            }).error(function(){
+                 // We must log in.
+                 // We Show the login prompt. Then try to log in:
+                $('#login').modal('show');
+            })
+        }
+
+        function loginSuccessfull(){
             setInterval($scope.refresh, 10000);
             Events.start();
         }
 
+        $scope.login = function () {
+            var username = $scope.currentFolder.username
+            var password = $scope.currentFolder.password
+
+            $http.post("rest/login?username="+username+"&password="+password).success(
```

Comment:
`urlbase` + `/rest/login...`

and some kind of URL quoting of the username/password.